[JIT] Move stuff out of class_type.cpp

jamesr66a opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack):
* #34140 [WIP][quantization] Use torchbind for Linear PackedParams
* #34339 [quantization] Make FP16 RNN use new prepack op
* #34098 [JIT] Introduce BuiltinOpFunction and integrate into torchbind
* #33921 [JIT] Virtualize Function
* **#33900 [JIT] Move stuff out of class_type.cpp**

These functions don't require any libtorch-specific functionality, so move them into the header so they're included in the ATen build

Differential Revision: [D20175874](https://our.internmc.facebook.com/intern/diff/D20175874)

Commit:
[JIT] Move stuff out of class_type.cpp

These functions don't require any libtorch-specific functionality, so move them into the header so they're included in the ATen build

Differential Revision: [D20146690](https://our.internmc.facebook.com/intern/diff/D20146690/)

[ghstack-poisoned]

Commit:
Update on "[JIT] Move stuff out of class_type.cpp"


These functions don't require any libtorch-specific functionality, so move them into the header so they're included in the ATen build

Differential Revision: [D20146690](https://our.internmc.facebook.com/intern/diff/D20146690/)

[ghstack-poisoned]

Commit:
Update on "[JIT] Move stuff out of class_type.cpp"


These functions don't require any libtorch-specific functionality, so move them into the header so they're included in the ATen build

Differential Revision: [D20146690](https://our.internmc.facebook.com/intern/diff/D20146690/)

[ghstack-poisoned]

zdevito reviewed (approved):
Let's put the implementations in type.cpp rather than the header to reduce incremental rebuild times but otherwise this is good.

Commit:
Update on "[JIT] Move stuff out of class_type.cpp"


These functions don't require any libtorch-specific functionality, so move them into the header so they're included in the ATen build

Differential Revision: [D20168124](https://our.internmc.facebook.com/intern/diff/D20168124)

[ghstack-poisoned]

Commit:
Update on "[JIT] Move stuff out of class_type.cpp"


These functions don't require any libtorch-specific functionality, so move them into the header so they're included in the ATen build

Differential Revision: [D20171610](https://our.internmc.facebook.com/intern/diff/D20171610)

[ghstack-poisoned]

Commit:
Update on "[JIT] Move stuff out of class_type.cpp"


These functions don't require any libtorch-specific functionality, so move them into the header so they're included in the ATen build

Differential Revision: [D20175874](https://our.internmc.facebook.com/intern/diff/D20175874)

[ghstack-poisoned]

Commit:
Update on "[JIT] Move stuff out of class_type.cpp"


These functions don't require any libtorch-specific functionality, so move them into the header so they're included in the ATen build

Differential Revision: [D20175874](https://our.internmc.facebook.com/intern/diff/D20175874)

[ghstack-poisoned]

Commit:
Update on "[JIT] Move stuff out of class_type.cpp"


These functions don't require any libtorch-specific functionality, so move them into the header so they're included in the ATen build

Differential Revision: [D20175874](https://our.internmc.facebook.com/intern/diff/D20175874)

[ghstack-poisoned]

Commit:
Update on "[JIT] Move stuff out of class_type.cpp"


These functions don't require any libtorch-specific functionality, so move them into the header so they're included in the ATen build

Differential Revision: [D20175874](https://our.internmc.facebook.com/intern/diff/D20175874)

[ghstack-poisoned]

Commit:
Update on "[JIT] Move stuff out of class_type.cpp"


These functions don't require any libtorch-specific functionality, so move them into the header so they're included in the ATen build

Differential Revision: [D20175874](https://our.internmc.facebook.com/intern/diff/D20175874)

[ghstack-poisoned]

Commit:
Update on "[JIT] Move stuff out of class_type.cpp"


These functions don't require any libtorch-specific functionality, so move them into the header so they're included in the ATen build

Differential Revision: [D20175874](https://our.internmc.facebook.com/intern/diff/D20175874)

[ghstack-poisoned]

Commit:
Update on "[JIT] Move stuff out of class_type.cpp"


These functions don't require any libtorch-specific functionality, so move them into the header so they're included in the ATen build

Differential Revision: [D20175874](https://our.internmc.facebook.com/intern/diff/D20175874)

[ghstack-poisoned]

Commit:
Update on "[JIT] Move stuff out of class_type.cpp"


These functions don't require any libtorch-specific functionality, so move them into the header so they're included in the ATen build

Differential Revision: [D20175874](https://our.internmc.facebook.com/intern/diff/D20175874)

[ghstack-poisoned]

Commit:
Update on "[JIT] Move stuff out of class_type.cpp"


These functions don't require any libtorch-specific functionality, so move them into the header so they're included in the ATen build

Differential Revision: [D20175874](https://our.internmc.facebook.com/intern/diff/D20175874)

[ghstack-poisoned]

Commit:
Update on "[JIT] Move stuff out of class_type.cpp"


These functions don't require any libtorch-specific functionality, so move them into the header so they're included in the ATen build

Differential Revision: [D20175874](https://our.internmc.facebook.com/intern/diff/D20175874)

[ghstack-poisoned]

Commit:
Update on "[JIT] Move stuff out of class_type.cpp"


These functions don't require any libtorch-specific functionality, so move them into the header so they're included in the ATen build

Differential Revision: [D20175874](https://our.internmc.facebook.com/intern/diff/D20175874)

[ghstack-poisoned]

Commit:
Update on "[JIT] Move stuff out of class_type.cpp"


These functions don't require any libtorch-specific functionality, so move them into the header so they're included in the ATen build

Differential Revision: [D20175874](https://our.internmc.facebook.com/intern/diff/D20175874)

[ghstack-poisoned]

Commit:
Update on "[JIT] Move stuff out of class_type.cpp"


These functions don't require any libtorch-specific functionality, so move them into the header so they're included in the ATen build

Differential Revision: [D20175874](https://our.internmc.facebook.com/intern/diff/D20175874)

[ghstack-poisoned]
